tests/scripts/git_skip_ci.sh

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env bash
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+set -eux
+
+# This implements the skip commands found here:
+# https://github.blog/changelog/2021-02-08-github-actions-skip-pull-request-and-push-workflows-with-skip-ci/
+
+# The script will return 1 if any of these are found in the HEAD commit message
+# headline
+
+if git log --format='%s' HEAD~1..HEAD | grep --quiet \
+    --regexp='\[skip ci\]' \
```

Comment:
As this is something new we are introducing, we could start with only one documented pattern rather than all these option variants, which makes a bit harder to manually grep on which commits in history didn’t run a full CI.

I am also a bit concerned on how we make sure that only exceptional changes that don’t really need CI will make use of this. As an extension, I wonder if Jenkins could somehow tag the PR with something when it detects the “skip ci” tag, so that it is obvious for reviewers that this didn’t run any checks, apart from lint.

Any idea on how to tackle these?

Reply:
Agreed on consolidating on a single trigger message like `[skip ci]` for now. At some level we'd need to rely on committers / mergers knowing that `[skip ci]` should only be applied to quick fixes and reverts to unbreak master which seems do-able. We could make it easier to tell without clicking by add a label like `CI was skipped` to the PR (through [GitHub Actions](https://github.com/actions/labeler) or (ideally) directly from Jenkins)

Reply:
Agreed. To me, that is a _must_ to have this PR in.

Another point, just to mention, is that we ideally would like to document it somewhere, so that it is clear for the community this is available and how to use it.